app/parse.py

```diff
@@ -8,8 +16,43 @@ class Quote:
     tags: list[str]
 
 
+def parse_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=quote_soup.select_one(".author").text,
+        tags=[tag_soup.text for tag_soup in quote_soup.select(".tags > a")]
+    )
+
+
+def get_page_quotes(url: str, quotes: list[Quote]) -> None:
+    page = requests.get(url).content
+    soup = BeautifulSoup(page, "html.parser")
+
+    quotes.extend(
+        [
+            parse_single_quote(quote_soup)
+            for quote_soup in soup.select(".quote")
+        ]
+    )
+
+    next_page = soup.select_one(".next > a")
+    if next_page is not None:
+        get_page_quotes(urljoin(URL, next_page["href"]), quotes)
+
+
+def get_all_quotes() -> list[Quote]:
+    quotes = []
+    get_page_quotes(URL, quotes)
```

Comment:
unnecessary method, u can create a list inside ur get_page_quotes method and return it